### PR TITLE
server: surface internal sessions with cluster setting value

### LINKED ISSUE #87009: filter out "$ api-v2-sql" from sessions page
Currently we're showing a lot of info coming from "$ api-v2-sql" on SQL Activity page. We should filter them out, specially on the Sessions page.

Jira issue: CRDB-19092

### PR DESCRIPTION
**Pull request by xinhaoz**
Closes #87009
Closes #84387

Previously, we include internal sessions in the API response
by default (which includes active execution info). This commit
aligns the surfacign  of internal stmts with the fingerprints
pages, returning internal sessions and active executions only
when `sql.stats.response.show_internal.enabled` is set to true.
This is with the exception of requests used to create internal
tables, such as `crdb_internal.cluster_sessions`, where we
continue to surface internal information.

The app name for sessions started from the UI using the SQL
over HTTP API have also been given the application name
`$ internal-console` to treat these sessions and their queries
as internal executions.

This commit also expands internal filtering for sql stats to
include queries that have app names starign with the
`DelegatedAppNamePrefix`, `$$`.

Release justification: low risk update to existing functionality
Release note (ui change): internal sessions and active executions
will only be surfaced in the UI when  the cluster setting
`sql.stats.response.show_internal.enabled` is set to true.

### CODE REVIEW

**Review by maryliag — APPROVED** [review]
:lgtm:

> Reviewed 13 of 13 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/87136#-:-NAo153N0cmQ9YoA1-i5:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @xinhaoz)

**Review by maryliag — APPROVED** [review]

**Review by rafiss — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/87136)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @maryliag, @rafiss, and @xinhaoz)

___
*[`-- commits` line 22 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/87136#-NAtakIT1EABR9OrbXiZ):*
nit: "starting"
___
*[`pkg/server/status.go` line 206 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/87136#-NAtb-oi62KrqFB-oG2u:-NAtb-oi62KrqFB-oG2v:bce0joh) ([raw file](https://github.com/cockroachdb/cockroach/blob/82a2a3db29522177e57a0bfdc4d283086a28349d/pkg/server/status.go#L206)):*
> ```Go
> 	// The empty username means "all sessions".
> 	showAll := reqUsername.Undefined()
> 	showInternal := SQLStatsShowInternal.Get(&b.st.SV) || req.IncludeInternal
> ```

this flag is a great idea - i imagine that in the future the UI could even include a checkbox to allow us to flip the flag more easily when debugging something. what do you think about adding a similar flag for the statement activity endpoint? (not going to block the PR on this, but just asking)
___
*[`pkg/ui/workspaces/cluster-ui/src/api/sqlApi.ts` line 17 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/87136#-NAtbq9o-QIP-iA_sSQD:-NAtbq9o-QIP-iA_sSQE:b-x2zoa6) ([raw file](https://github.com/cockroachdb/cockroach/blob/82a2a3db29522177e57a0bfdc4d283086a28349d/pkg/ui/workspaces/cluster-ui/src/api/sqlApi.ts#L17)):*
> ```XML
>   execute?: boolean;
>   timeout?: string; // Default 5s
>   application_name?: string; // Defaults to '$ internal'
> ```

should this say `defaults to '$ internal-console-http'`? (that is, unless you change it per my comment below.)
___
*[`pkg/ui/workspaces/cluster-ui/src/api/sqlApi.ts` line 64 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/87136#-NAtbGfm9I4Mc8cZ1iyr:-NAtbGfm9I4Mc8cZ1iys:b-9vygke) ([raw file](https://github.com/cockroachdb/cockroach/blob/82a2a3db29522177e57a0bfdc4d283086a28349d/pkg/ui/workspaces/cluster-ui/src/api/sqlApi.ts#L64)):*
> ```XML
> 
> export const SQL_API_PATH = "/api/v2/sql/";
> export const INTERNAL_SQL_API_APP = "$ internal-console-http";
> ```

i don't know if i fully agree with making this "internal." there's a major difference here in that the queries in the sql-over-http endpoint are initiated outside of CRDB and the db has no knowledge of what's being executed. this is unlike every other internal query, where the query itself is in the CRDB source code. maybe using `DelegatedAppNamePrefix` makes more sense here

also, it feels like setting this default should be done in the CRDB server, not by the client.

**Review by xinhaoz — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/87136)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @maryliag, @rafiss, and @xinhaoz)

___
*[`pkg/server/status.go` line 206 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/87136#-NAtb-oi62KrqFB-oG2u:-NAtlIIc52s-n-VEJHjz:btzw45m) ([raw file](https://github.com/cockroachdb/cockroach/blob/82a2a3db29522177e57a0bfdc4d283086a28349d/pkg/server/status.go#L206)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

this flag is a great idea - i imagine that in the future the UI could even include a checkbox to allow us to flip the flag more easily when debugging something. what do you think about adding a similar flag for the statement activity endpoint? (not going to block the PR on this, but just asking)
</blockquote></details>

Yup, that's the eventual goal! This is the tracking issue for that: https://github.com/cockroachdb/cockroach/issues/87200
___
*[`pkg/ui/workspaces/cluster-ui/src/api/sqlApi.ts` line 64 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/87136#-NAtbGfm9I4Mc8cZ1iyr:-NAtlagG1IgzdvLJ60xp:b-19e2gn) ([raw file](https://github.com/cockroachdb/cockroach/blob/82a2a3db29522177e57a0bfdc4d283086a28349d/pkg/ui/workspaces/cluster-ui/src/api/sqlApi.ts#L64)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

i don't know if i fully agree with making this "internal." there's a major difference here in that the queries in the sql-over-http endpoint are initiated outside of CRDB and the db has no knowledge of what's being executed. this is unlike every other internal query, where the query itself is in the CRDB source code. maybe using `DelegatedAppNamePrefix` makes more sense here

also, it feels like setting this default should be done in the CRDB server, not by the client.
</blockquote></details>

Ah good point. The app name here is only for queries executed by the console, the actual default app name for the sql api is `$ api-v2-sql` which is filled in server-side. The goal here was just to get these queries bucked into the behaviour we have for filtering internal sessions without doing extra work - all the queries executed in the sql http api will have the delegated prefix.

Before making this change, just want to make sure it's alright that a session itself has an app name with the delegated prefix, or if you think it makes more sense to give it a regular application name here and add it to the list of app names to be filtered out with the flag. Thanks for reviewing Rafi!

**Review by rafiss — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/87136)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @maryliag, @rafiss, and @xinhaoz)

___
*[`pkg/server/status.go` line 206 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/87136#-NAtb-oi62KrqFB-oG2u:-NAuKlQv73HDVr5L4_0f:b-7m74ox) ([raw file](https://github.com/cockroachdb/cockroach/blob/82a2a3db29522177e57a0bfdc4d283086a28349d/pkg/server/status.go#L206)):*
<details><summary><i>Previously, xinhaoz (Xin Hao Zhang) wrote…</i></summary><blockquote>

Yup, that's the eventual goal! This is the tracking issue for that: https://github.com/cockroachdb/cockroach/issues/87200
</blockquote></details>

nice!
___
*[`pkg/ui/workspaces/cluster-ui/src/api/sqlApi.ts` line 64 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/87136#-NAtbGfm9I4Mc8cZ1iyr:-NAuJMuA7OZO50CNATDq:b-j1jae5) ([raw file](https://github.com/cockroachdb/cockroach/blob/82a2a3db29522177e57a0bfdc4d283086a28349d/pkg/ui/workspaces/cluster-ui/src/api/sqlApi.ts#L64)):*
<details><summary><i>Previously, xinhaoz (Xin Hao Zhang) wrote…</i></summary><blockquote>

Ah good point. The app name here is only for queries executed by the console, the actual default app name for the sql api is `$ api-v2-sql` which is filled in server-side. The goal here was just to get these queries bucked into the behaviour we have for filtering internal sessions without doing extra work - all the queries executed in the sql http api will have the delegated prefix.

Before making this change, just want to make sure it's alright that a session itself has an app name with the delegated prefix, or if you think it makes more sense to give it a regular application name here and add it to the list of app names to be filtered out with the flag. Thanks for reviewing Rafi!
</blockquote></details>

yeah good point about making a real session with the delegated prefix. i don't think we should do that

ah ok thanks for explaining. if the queries being sent here are all crafted in the DB Console code, then i am open to the `$ internal-console-http` name. the main thing is to make sure nothing user-generated gets here.

i think a name like `$ console` would work too, and it matches what the CLI does (it gets an app name of `$ cockroach sql`). but if that's too much extra work, no need.

**Comment by xinhaoz** [comment]
*[`pkg/ui/workspaces/cluster-ui/src/api/sqlApi.ts` line 64 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/87136#-NAtbGfm9I4Mc8cZ1iyr:-NAuTW7K0jAdKaxX43nO:b-tz69ob) ([raw file](https://github.com/cockroachdb/cockroach/blob/82a2a3db29522177e57a0bfdc4d283086a28349d/pkg/ui/workspaces/cluster-ui/src/api/sqlApi.ts#L64)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

yeah good point about making a real session with the delegated prefix. i don't think we should do that

ah ok thanks for explaining. if the queries being sent here are all crafted in the DB Console code, then i am open to the `$ internal-console-http` name. the main thing is to make sure nothing user-generated gets here.

i think a name like `$ console` would work too, and it matches what the CLI does (it gets an app name of `$ cockroach sql`). but if that's too much extra work, no need.


</blockquote></details>

Ah you know what actually I based the sessions filtering off of whether the executor was internal, not the app name, and since these queries are all executed using an internal executor they'll all be filtered accordingly. So setting the name to `$ console` for these reqs should work fine here if we go with that, but we would need to revisit this if the UI based sql shell ever comes into existence. @maryliag any thoughts here on whether I should change the filtering to be based on application name instead and whether we should also keep the change here on filtering delegated queries in the sql stats side? Or maybe can continue on this current route and defer to when we actually want to execute user queries through this API + further discussions with obs-infra folks.

**Comment by xinhaoz** [comment]
*[`pkg/ui/workspaces/cluster-ui/src/api/sqlApi.ts` line 64 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/87136#-NAtbGfm9I4Mc8cZ1iyr:-NAuViwAEMAX42KwDGwD:b-dga8v8) ([raw file](https://github.com/cockroachdb/cockroach/blob/82a2a3db29522177e57a0bfdc4d283086a28349d/pkg/ui/workspaces/cluster-ui/src/api/sqlApi.ts#L64)):*
<details><summary><i>Previously, xinhaoz (Xin Hao Zhang) wrote…</i></summary><blockquote>

Ah you know what actually I based the sessions filtering off of whether the executor was internal, not the app name, and since these queries are all executed using an internal executor they'll all be filtered accordingly. So setting the name to `$ console` for these reqs should work fine here if we go with that, but we would need to revisit this if the UI based sql shell ever comes into existence. @maryliag any thoughts here on whether I should change the filtering to be based on application name instead and whether we should also keep the change here on filtering delegated queries in the sql stats side? Or maybe can continue on this current route and defer to when we actually want to execute user queries through this API + further discussions with obs-infra folks.
</blockquote></details>

Above comment was made with teh assumptions we'd want to track ui sql shell queries as non-internal.

**Review by maryliag — APPROVED** [review]
:lgtm:

> Reviewed 6 of 6 files at r2, 12 of 16 files at r3, 1 of 1 files at r4.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/87136#-:-NBJhfes79A3p1jE3APs:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @maryliag, @rafiss, and @xinhaoz)

___
*[`pkg/ui/workspaces/cluster-ui/src/api/sqlApi.ts` line 64 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/87136#-NAtbGfm9I4Mc8cZ1iyr:-NBJhQenCjpOL2PtcByG:b-gdpwn8) ([raw file](https://github.com/cockroachdb/cockroach/blob/82a2a3db29522177e57a0bfdc4d283086a28349d/pkg/ui/workspaces/cluster-ui/src/api/sqlApi.ts#L64)):*
<details><summary><i>Previously, xinhaoz (Xin Hao Zhang) wrote…</i></summary><blockquote>

Above comment was made with teh assumptions we'd want to track ui sql shell queries as non-internal.
</blockquote></details>

Discussed offline, the filtering will be done by app name/session name instead of executor type.
___
*[`pkg/ui/workspaces/cluster-ui/src/api/sqlApi.ts` line 91 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/87136#-NBJh8AM92_t8c9oQOlr:-NBJh8AM92_t8c9oQOls:b-l7in4q) ([raw file](https://github.com/cockroachdb/cockroach/blob/0a71b12edf976aa1187f82128f0855fa6802c8a7/pkg/ui/workspaces/cluster-ui/src/api/sqlApi.ts#L91)):*
> ```XML
>  * @param req execution request details
>  */
> export function executeInternalSql<RowType>(
> ```

nice! that will help make things clear!

**Review by rafiss — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/87136)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @maryliag, @rafiss, and @xinhaoz)

___
*[`pkg/ui/workspaces/cluster-ui/src/api/sqlApi.ts` line 86 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/87136#-NBNsnNxEN9ntRPS5ZpU:-NBNsnO39S_ek9puIC1m:by6pdea) ([raw file](https://github.com/cockroachdb/cockroach/blob/0a71b12edf976aa1187f82128f0855fa6802c8a7/pkg/ui/workspaces/cluster-ui/src/api/sqlApi.ts#L86)):*
> ```XML
>  * the app name set to the internal sql api app name above.
>  * Note that technically all SQL executed over this API is
>  * executed as internal, but we make this distinction for when
> ```

i may not be following this comment. when we execute user-initiated queries, we should not use the `$ internal` prefix, but it looks like the `else` branch is always adding `$ internal`

**Comment by xinhaoz** [comment]
*[`pkg/ui/workspaces/cluster-ui/src/api/sqlApi.ts` line 86 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/87136#-NBNsnNxEN9ntRPS5ZpU:-NBNtuEgD83ssR4YGXEN:bj2qzqh) ([raw file](https://github.com/cockroachdb/cockroach/blob/0a71b12edf976aa1187f82128f0855fa6802c8a7/pkg/ui/workspaces/cluster-ui/src/api/sqlApi.ts#L86)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

i may not be following this comment. when we execute user-initiated queries, we should not use the `$ internal` prefix, but it looks like the `else` branch is always adding `$ internal`
</blockquote></details>

This function `executeInternalSql` should only be used to execute queries constructed from within the console and not user initiated. There's the general `executeSql` function. This comment is moreso to explain that despite these function names, _all_ queries executed by the API will use an internal executor.

**Review by rafiss — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/87136)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @maryliag, @rafiss, and @xinhaoz)

___
*[`pkg/ui/workspaces/cluster-ui/src/api/sqlApi.ts` line 86 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/87136#-NBNsnNxEN9ntRPS5ZpU:-NBO05_23ogW4ocR7AHm:b80i42m) ([raw file](https://github.com/cockroachdb/cockroach/blob/0a71b12edf976aa1187f82128f0855fa6802c8a7/pkg/ui/workspaces/cluster-ui/src/api/sqlApi.ts#L86)):*
<details><summary><i>Previously, xinhaoz (Xin Hao Zhang) wrote…</i></summary><blockquote>

This function `executeInternalSql` should only be used to execute queries constructed from within the console and not user initiated. There's the general `executeSql` function. This comment is moreso to explain that despite these function names, _all_ queries executed by the API will use an internal executor.
</blockquote></details>

ok, sounds good. i don't quite understand the "but we make this distinction for when we may want to execute user queries." part of the sentence. i'll leave it up to you if you want to reword it

**Comment by xinhaoz** [comment]
TFTR!
bors r+

**Comment by xinhaoz** [comment]
blathers backport 22.2

**Pull request merged**
